Automattic/wp-calypso
# Domains: Change signup destination to `setup-site` flow if creating a new site

#### Changes proposed in this Pull Request
Redirects the user to the `setup-site` signup flow after checkout if the "New site" option was selected on the "Choose how to use your domain" step.
This PR is a part of the domain-only flow redesign described in g1kxLPDmNsSHje0aA2Qjme-fi-728%3A3920.

#### Testing instructions
- Browse to `/start/domain`;
- Try to use any available domain on the search step;
- Select the "New site" option on the "Choose how to use your domain" step;
- Select any theme and plan from the next steps;
- Confirm that after checking out you'll get redirected to the `setup-site` flow (Intent capture - "Where will you start?" page)
